### PR TITLE
fix: closed trades timezone consistency

### PR DESCRIPTION
**Pull request by uhliksk**
## Description

- Fix data inconsistencies caused by using various timezone settings on client and server side. On the server side dates will be always stored in UTC format regardless of docker container timezone.
- Fix wrong periods of time reported on client in client's timezone is other than UTC. Client will report it's timezone to server to get proper range of data from database based on local timezone.
- Fix wrong first day of week reported on client with locale set to other day than sunday. Client will report it's locale to server to get proper range of data from database based on the locale.

## Related Issue

#449

## Motivation and Context

Closed trades were reporting different values on different places in application when day, week or month were selected as reported period of time.

## How Has This Been Tested?

Docker container is using UTC time in default settings thus the change will not affect default configurations. In configurations with custom docker container time zone there may be temporary change in order of transactions stored in database if UTC time predeces the local time as the key was previously also generated including the time zone data.

Tested using custom `TZ` variable to change default docker container time zone. On the client side tested on system with time zone other than UTC.

### CODE REVIEW

**Comment by habibalkhabbaz** [comment]
Thanks @uhliksk
It seems a good PR!
However, I think the tests failed. Are you planned to fix them? If not, just let me know and I will be happy to fix and to keep things easier for @chrisleekr when he have time to check.

**Comment by uhliksk** [comment]
@habibalkhabbaz Oops, I didn't checked the tests. You can fix them, of course. Thank you.

**Comment by habibalkhabbaz** [comment]
Hello @uhliksk 
I've finished fixing the test but unfortunately, I can't commit due to permission.
I don't know if this can be done from your side or from @chrisleekr side

**Comment by chrisleekr** [comment]
> Hello @uhliksk I've finished fixing the test but unfortunately, I can't commit due to permission. I don't know if this can be done from your side or from @chrisleekr side
> 
> ```
> remote: Permission to uhliksk/binance-trading-bot.git denied to habibalkhabbaz.
> ```

@habibalkhabbaz I think @uhliksk should add you as a collaborator in his forked repository.

**Comment by chrisleekr** [comment]
@uhliksk Thanks for your contributions.

As soon as @habibalkhabbaz commits tests, I will review and merge in.

**Comment by uhliksk** [comment]
Hello @habibalkhabbaz

I've sent you an invite for `uhliksk/binance-trading-bot`.

**Comment by habibalkhabbaz** [comment]
Thanks @uhliksk @chrisleekr 
Tests has been added.

File: app/cronjob/trailingTrade/step/handle-open-orders.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ const execute = async (logger, rawData) => {
           data.accountInfo = await updateAccountInfo(
             logger,
             balances,
-            moment().utc().format()
+            moment().utc().toISOString()
```
Comment (chrisleekr) [inline app/cronjob/trailingTrade/step/handle-open-orders.js:156]: `moment().utc().toISOString()` and `moment().toISOString()` are same format. 
Screen Shot 2022-08-07 at 17 14 44

I am doing some changes, let me update it.

Comment (chrisleekr) [reply app/cronjob/trailingTrade/step/handle-open-orders.js]: Updated.

**Comment by chrisleekr** [comment]
@uhliksk @habibalkhabbaz 

Hi guys,

I pushed some changes. 
https://github.com/chrisleekr/binance-trading-bot/pull/454/commits/40dc23e08b7523d06af589c8cfd7f51817155dc0
Please review and let me know what you think.

So the principle I get from this PR is,

1. Anything saved into the database should be a UTC Date instance. 
  i.e.
  ```
  moment().utc().toDate()
  ```
2. Slack message will be the docker container's local date time.
  i.e.
  ```
  moment().format()
  ```

3. Override data `actionAt` should be UTC ISO string format. 
  i.e.
  ```
  moment().toISOString()
  ```

Did I understand correctly? Let me know if I understand wrong.

And one suggestion for @uhliksk 
It looks like you have not installed husky properly because it has forced lint rule for branch name.
You must see this error when you commit or push
```
Branch name lint fail! Branch "fix-closed-trades-timezone-consistency" must contain a separator "/".
``` 
And I also found some lint is not applied. Are you using VSCode? 
If not, prefer to use VSCode as the project is optimised for it.

**Comment by uhliksk** [comment]
@uhliksk Thank you for the suggestion. I'll install and use VSCode.

**Comment by habibalkhabbaz** [comment]
@chrisleekr 
I see it's fine to merge in if @uhliksk not have any issue.

**Comment by uhliksk** [comment]
> @chrisleekr I see it's fine to merge in if @uhliksk not have any issue.

No issue. Fine to merge for me.

**Comment by chrisleekr** [comment]
Very good.
I will merge in.

@uhliksk @habibalkhabbaz

**Pull request merged**